### PR TITLE
Update app secrets article for beta7

### PR DESCRIPTION
**Pull request by danroth27**

### CODE REVIEW

**Comment by danroth27** [comment]
@Rick-Anderson @glennc

File: aspnet/security/app-secrets.rst [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
+Apps frequently require secrets, such as a client ID for OAuth or database connection strings with credentials. These passwords and other sensitive data should never be kept in source code or added to configuration files inside your project's source tree, as they can easily be accessed by untrusted parties.
+
+This articles shows how you can manage and access secrets in your local development environment without resorting to hardcoded values or local configuration files. You can instead use the :doc:`configuration </fundamentals/configuration>` system to read these values from environment variables or from values stored using the Secret Manager tool. The Secret Manager tool can be used to manage and store user secrets while integrating seamlessly with the configuration system.
```
Comment (glennc) [inline aspnet/security/app-secrets.rst]: Article

File: aspnet/security/app-secrets.rst [hunk 2]
```diff
@@ -1,9 +1,11 @@
 Safe Storage of Application Secrets
 ===================================
 
-By `Rick Anderson`_
+By `Rick Anderson`_, `Daniel Roth`_
 
-This tutorial shows how your application can securely store and access secrets in the local development environment. The most important point is you should never store passwords or other sensitive data in source code, and you shouldn't use production secrets in development and test mode. The secret manager tool was written to help prevent sensitive data from being checked into source control. The :doc:`/fundamentals/configuration` system that is used by default in DNX based apps can read secrets stored with the secret manager tool described in this article.
+Apps frequently require secrets, such as a client ID for OAuth or database connection strings with credentials. These passwords and other sensitive data should never be kept in source code or added to configuration files inside your project's source tree, as they can easily be accessed by untrusted parties.
```
Comment (glennc) [inline aspnet/security/app-secrets.rst]: Should this be softened or expanded upon? Talk about checking in a secret that makes its way to GitHub?

Comment (Rick-Anderson) [inline aspnet/security/app-secrets.rst]: I like the original better. The new version is missing the "you shouldn't use production secrets in development and test mode." - this isn't directly related to the secrets manager, but we should help our customers fall into the pit of security success. @blowdart actually came up with the original. "The most important point" is attention grabbing.

File: aspnet/security/app-secrets.rst [hunk 3]
```diff
@@ -14,76 +16,86 @@ In this article:
...
-- Linux: ``~/.microsoft/usersecrets/<applicationId>/secrets.json``
-- Mac: ``~/.microsoft/usersecrets/<applicationId>/secrets.json``
+Then add the user secrets configuration source:
 
-The ``applicationId`` comes from the the *project.json* file and is arbitrary, but should be unique unless you have a reason for it not to be. The following markup shows a portion of the *project.json* file with the ``applicationId`` highlighted:
+.. code-block:: c#
 
-.. code-block:: json
-  :emphasize-lines: 3
+  builder.AddUserSecrets();
```
Comment (glennc) [inline aspnet/security/app-secrets.rst]: builder? :)

File: aspnet/security/app-secrets.rst [hunk 4]
```diff
@@ -14,76 +16,86 @@ In this article:
...
+You can now access user secrets via the configuration API:
 
-  {
-    "webroot": "wwwroot",
-    "userSecretsId": "aspnet5-WebApplication1-f7fd3f56-2899-4eea-a88e-673d24bd7090",
-    "version": "1.0.0-*"
-  }
+.. code-block:: c#
+
+  string testConfig = configuration.Get("MySecret");
```
Comment (glennc) [inline aspnet/security/app-secrets.rst]: configuration["MySecret"]

**Comment by glennc** [comment]
I didn't really get a feel from this on when I should or shouldn't use the UserSecrets features. But I am ok with the changes. :shipit:

**Comment by Rick-Anderson** [comment]
@glennc - agreed on you don't know when. http://www.asp.net/identity/overview/features-api/best-practices-for-deploying-passwords-and-other-sensitive-data-to-aspnet-and-azure does a much better job. From that:

Tutorials frequently show sensitive data in source code, hopefully with a caveat that you should never store sensitive data in source code. For example, my ASP.NET MVC 5 app with SMS and email 2FA tutorial shows the following in the web.config file:

**Comment by Rick-Anderson** [comment]
Nice improvements  :shipit:

**Pull request merged**